sympy/solvers/solveset.py

```diff
@@ -999,6 +1007,282 @@ def _solveset(f, symbol, domain, _check=False):
     return result
 
 
+def _is_modular(f, symbol):
+    """
+    Helper function to check below mentioned types of modular equations.
+    ``A - Mod(B, C) = 0``
+
+    A -> This can or cannot be a function of symbol.
+    B -> This is surely a function of symbol.
+    C -> It is an integer.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The equation to be checked.
+
+    symbol : Symbol
+        The concerned variable for which the equation is to be checked.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, exp, Mod
+    >>> from sympy.solvers.solveset import _is_modular as check
+    >>> x, y = symbols('x y')
+    >>> check(Mod(x, 3) - 1, x)
+    True
+    >>> check(Mod(x, 3) - 1, y)
+    False
+    >>> check(Mod(x, 3)**2 - 5, x)
+    False
+    >>> check(Mod(x, 3)**2 - y, x)
+    False
+    >>> check(exp(Mod(x, 3)) - 1, x)
+    False
+    >>> check(Mod(3, y) - 1, y)
+    False
+    """
+
+    if not f.has(Mod):
+        return False
+
+    # extract modterms from f.
+    modterms = list(f.atoms(Mod))
+
+    return (len(modterms) == 1 and  # only one Mod should be present
+            modterms[0].args[0].has(symbol) and  # B-> function of symbol
+            modterms[0].args[1].is_integer and  # C-> to be an integer.
+            any(isinstance(term, Mod)
+            for term in list(_term_factors(f)))  # free from other funcs
+            )
+
+
+def _invert_modular(modterm, rhs, n, symbol):
+    """
+    Helper function to invert modular equation.
+    ``Mod(a, m) - rhs = 0``
+
+    Generally it is inverted as (a, ImageSet(Lambda(n, m*n + rhs), S.Integers)).
+    More simplified form will be returned if possible.
+
+    If it is not invertible than modterm, rhs is returned.
+
+    Following cases arise:
+    1. a is x -> m*n + rhs is the required solution.
+    2. a is instance of Add -> we try to find two independent parts
+    of a and the symbol independent gets tranferred to other
+    side and again the solveset is called on the symbol dependent part.
+    3. a is instance of Mul -> same as we done in Add we separate out
+    the dependent and independent parts and tranfer the independent
+    part to rhs using invert here.
+    4. a is instance of Pow i.e a = base**exp -> 2 cases arise as following:
+    - a is (x_indep)**(x_dep) -> Firstly remainder is evaluated with the help
+    of discrete_log function and then the least period is being found out
+    with the help of totient function. For reference:
+    (https://en.wikipedia.org/wiki/Euler's_theorem)
+    - a is (x_dep)**(x_indep) -> Firstly all primitive solutions are being found
+    with the help of nthroot_mod function if possible. And then the general
+    solution m*n + rem is being returned where rem belongs to solutions returned
+    by nthroot_mod function.
```

Comment:
The language of this documentation is not up-to sympy standards, this needs improvement. Please have a look at `sympy/solvers/pde.py` for motivation.

Reply:
I went through `sympy/solvers/pde.py`. Is it okay now?

Reply:
Is it okay now @aktech ?